src/Workspaces/Core/MSBuild/MSBuild/Build/ProjectBuildManager.cs

```diff
@@ -197,31 +197,42 @@ public void EndBatchBuild()
         {
             Debug.Assert(BatchBuildStarted);
 
-            var targets = new[] { TargetNames.Compile, TargetNames.CoreCompile };
+            var requiredTargets = new[] { TargetNames.Compile, TargetNames.CoreCompile };
+            var optionalTargets = new[] { TargetNames.DesignTimeMarkupCompilation };
```

Comment:
💡 It looks like if we set `BuildingInsideVisualStudio` to true, this will run automatically

Reply:
I'll merge in the changes from https://github.com/dotnet/roslyn/pull/57219 and close that one out.

Reply:
Actually, after testing locally, setting BuildingInsideVisualStudio breaks the `blazor` and `blazorwasm` scenarios. Will have to dig a bit to understand why.

Reply:
@sharwell Setting `BuildingInsideVisualStudio` causes the RazorSourceGenerator.razorencconfig file to be added as an AnalyzerConfig, which then adds `build_property.SuppressRazorSourceGenerator = true` to the configuration. I noticed Roslyn's feature layer has a class dedicated to removing this config file from the Solution (https://github.com/dotnet/roslyn/blob/f5b6c715a742c56b7cc672e47385508fb4df98cc/src/Features/Core/Portable/Workspace/CompileTimeSolutionProvider.cs). I think we should take the change as is until Razor has sorted out their design time build.

Reply:
@davidwengier Do you know the status of the Razor design time build work?

Reply:
There won't be any change Razor design time code gen in the short term.
Additionally the last time someone tried to change whether BuildingInsideVisualStudio was set it broke dotnet watch and various VS Code things, from memory.